changelog/_unreleased/2024-07-05-made-navigation-route-validate-function-decoratable.md

```diff
@@ -0,0 +1,10 @@
+---
+title: Made NavigationRoute validate function decorateable
+date: 2024-07-05
+area: core
+---
+
+# Core
+* Added the new `NavigationRouteValidate` Event to make it possbile to set the NavigationRoute state valid based on logic in event subscribers.
+* Moved the SalesChannel category id validation check to the `NavigationRouteValidateSubscriber` using the new `NavigationRouteValidateEvent`.
+* Moved utility functions from the `NavigationRoute` to a generic `CategoryService` so this functions can be reused.
```

Comment:
Please start your changed with one of those keywords: Added, Changed, Deprecated, Removed
As written [in this adr](https://github.com/shopware/shopware/blob/trunk/adr/2020-08-03-implement-new-changelog.md)